matrix-sync.c

```diff
@@ -157,6 +157,12 @@ static void matrix_sync_room(const gchar *room_id,
                 timeline_object, "events");
     if(timeline_array != NULL)
         _parse_room_event_array(conv, timeline_array, FALSE);
+
+    /* parse the ephemeral events */
+    ephemeral_object = matrix_json_object_get_object_member(room_data, "ephemeral");
+    ephemeral_array = matrix_json_object_get_array_member(ephemeral_object, "events");
+    if(ephemeral_array != NULL)
+        _parse_room_event_array(conv, ephemeral_array, TRUE);
```

Comment:
typing events aren't really state events, so this `TRUE` looks wrong at first sight.

I think that using the state table to track the typing state is a valid hack, so that's fine, but a comment here to explain what's going on would be helpful.